libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/BannerInstructionEvent.kt

```diff
@@ -12,16 +11,18 @@ internal class BannerInstructionEvent {
     var latestBannerInstructions: BannerInstructions? = null
         private set
 
-    fun isOccurring(routeProgress: RouteProgress) = updateCurrentBanner(routeProgress)
+    fun isOccurring(routeProgress: RouteProgress): Boolean = updateCurrentBanner(routeProgress)
 
     fun invalidateLatestBannerInstructions() {
         latestBannerInstructions = null
     }
 
-    private fun updateCurrentBanner(routeProgress: RouteProgress) {
+    private fun updateCurrentBanner(routeProgress: RouteProgress): Boolean {
         bannerInstructions = routeProgress.bannerInstructions
-        ifNonNull(bannerInstructions) {
-            latestBannerInstructions = it
+        if (bannerInstructions != null && bannerInstructions!! != latestBannerInstructions) {
```

Comment:
are complex objects comparable by contents here?

Reply:
Yeah, this is Kotlin notation but underneath is `equals`.